WeakAurasOptions/WeakAurasOptions.lua

```diff
@@ -1256,6 +1240,47 @@ function WeakAuras.DeleteOption(data)
   end
 end
 
+StaticPopupDialogs["WEAKAURAS_CONFIRM_DELETE"] = {
+  text = L["Are you sure you want to delete?"],
+  button1 = L["Yes"],
+  button2 = L["No"],
+  OnAccept = function(self)
+    if self.data then
+      for _, auraData in pairs(self.data.toDelete) do
+        WeakAuras.DeleteOption(auraData)
+      end
+      if self.data.parents then
+        for id in pairs(self.data.parents) do
+          local parentData = WeakAuras.GetData(id)
+          local parentButton = WeakAuras.GetDisplayButton(id)
+          WeakAuras.UpdateGroupOrders(parentData)
+          if(#parentData.controlledChildren == 0) then
+            parentButton:DisableExpand()
+          else
+            parentButton:EnableExpand()
+          end
+          parentButton:SetNormalTooltip()
+        end
+      end
+      WeakAuras.SortDisplayButtons()
+    end
+  end,
+  OnCancel = function(self)
+    self.data = nil
+  end,
+  showAlert = true,
+  whileDead = true,
+  preferredindex = STATICPOPUP_NUMDIALOGS,
+}
+
+function WeakAuras.ConfirmDelete(toDelete, parents)
+  if toDelete then
+    StaticPopup_Show("WEAKAURAS_CONFIRM_DELETE", "", "", {toDelete = toDelete, parents = parents})
```

Comment:
The message should tell the user what they are deleting, that this the group name, aura name, or for any other case properly somthing like "deleting x auras". 

The buttons in message boxes should never be "yes"/"no", but always verbs indicating the action that is about to be taken.

In this case: "Keep" or "Cancel" and "Delete".